mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulPuzzleGameScreen.kt

```diff
@@ -39,7 +39,9 @@ fun StatefulPuzzleGameScreen(
   val speechFacade = LocalSpeechFacade.current
 
   val scope = rememberCoroutineScope()
-  val puzzle = chessFacade.puzzle(uid = puzzleId) ?: Puzzle()
+  val puzzle =
+      remember(chessFacade) { chessFacade.puzzle(uid = puzzleId) }.collectAsState(Puzzle()).value
+          ?: Puzzle()
```

Comment:
Just fyi : you could probably have done something like

```kotlin
val puzzle by remember(chessFacade) {
  chessFacade.puzzle(uid = puzzle).map { it ?: Puzzle() }
}.collectAsState(Puzzle())
```

to use the property delegates of `State`.

Reply:
https://github.com/epfl-SDP/android/pull/450/commits/49543540a19b86a0b1a17c1e91e3217b4a346e72